File: Tareas/tarea15.rb
puts "Azul Carolina Martínez Cabañas"
puts "1868435					 ITS"
puts "Periodo Enero-Junio 2020"
puts "Ing. Ismael Gutiérrez Pimentel"
puts("Suma de matrices")
print "Ingresar el número de filas de la matriz A: "
mA = gets().to_i
print "Ingresar el número de columnas de la matriz A: "
nA = gets().to_i
print "Ingresar el número de filas de la matriz B: "
mB = gets().to_i
print "Ingresar el número de columnas de la matriz B: "
nB = gets().to_i
if nA != nB || mA != mB
    abort("Las dimensiones de las matrices no son válidas")
else
    
    matA = Array.new(mA){Array.new(nA)}
    matB = Array.new(mB){Array.new(nB)}
    for i in(0..(mA - 1))
        for j in(0..(nA - 1))
            print "Ingresar los números que pertenecen a la matriz A[#{i+1}][#{j+1}] :"
            matA[i][j]=gets().to_i
        end
    end
    for i in(0..(mB - 1))
        for j in(0..(nB - 1))
            print "Ingresar los números que pertenecen a la matriz B[#{i+1}][#{j+1}] :"
            matB[i][j]=gets().to_i
        end
    end
    mensaje = ""
    suma = Array.new(mA){Array.new(nA)}
    mensaje=mensaje+"La suma de ambas matrices es \n"
    for i in(0..(mA-1))
        for j in(0..(nA-1))
            suma[i][j]= matA[i][j]+matB[i][j]
            mensaje= mensaje+suma[i][j].to_s+" "
        end
        mensaje=mensaje+"\n"
    end
    print"#{mensaje}"
end

File: Tareas/tarea15.java
import javax.swing.*;

public class Tarea15 {
    public static void main(String[] args) {
        JOptionPane.showMessageDialog(null, "Azul Carolina Martínez Cabañas\n1868435 ITS\nPeriodo Enero-Junio 2020\nIng. Ismael Gutiérrez Pimentel");
        int mA = Integer.parseInt(JOptionPane.showInputDialog("Suma de matrices\nIngresar el número de filas de la matriz A:"));
        int nA = Integer.parseInt(JOptionPane.showInputDialog("Ingresar el número de columnas de la matriz A:"));
        int mB = Integer.parseInt(JOptionPane.showInputDialog("Ingresar el número de filas de la matriz B:"));
        int nB = Integer.parseInt(JOptionPane.showInputDialog("Ingresar el número de columnas de la matriz B:"));
        if(mA != mB || nA != nB){
            JOptionPane.showMessageDialog(null, "Las dimensiones de las matrices no son válidas");
            System.exit(0);
        }else{
            int [][] matA = new int[mA][nA];
            int [][] matB = new int[mB][nB];
            for (int i = 0; i <=(mA-1) ; i++) {
                for (int j = 0; j <=(nA -1); j++) {
                    matA[i][j] = Integer.parseInt(JOptionPane.showInputDialog("Ingresar el numero de la matriz A "));
                }
            }
            for (int i = 0; i <=(mB -1) ; i++) {
                for (int j = 0; j <= (nB -1); j++) {
                    matB[i][j] = Integer.parseInt(JOptionPane.showInputDialog("Ingresar el numero de la matriz B"));
                }
            }
            int[][] suma = new int[mA][nA];
            String mensaje = "La suma de ambas matrices es\n";
            for (int i = 0; i <=(mA-1) ; i++) {
                for (int j = 0; j <=(nA -1); j++) {
                    suma[i][j] = matA[i][j] + matB[i][j];
                    mensaje += suma[i][j];
                    mensaje += " ";
                }
                mensaje += "\n";
            }
            JOptionPane.showMessageDialog(null, mensaje);
        }
    }
}